src/main/java/com/gradle/develocity/bamboo/Utils.java

```diff
@@ -0,0 +1,30 @@
+package com.gradle.develocity.bamboo;
+
+import com.atlassian.bamboo.v2.build.BuildContext;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+class Utils {
+
+    private Utils() {
+    }
+
+    static Set<String> vcsRepoUrls(BuildContext buildContext) {
```

Comment:
I'd make this method part of `DevelocityPreBuildAction` class. Introducing this `Utils` class seems very artificial and it's better to bind this logic to some domain entity. 

If extracting, I'd extract the whole vcs decision logic separately and use it as a component in `DevelocityPreBuildAction`; So either all there via methods or all out in it's own entity.